Sources/Purchasing/StoreKit2/StoreKit2TransactionFetcher.swift

```diff
@@ -90,15 +116,109 @@ extension StoreKit.VerificationResult where SignedType == StoreKit.Transaction {
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(String(transaction.id), error))
+            return nil
+        }
+    }
+
+}
+
+// MARK: - Private
+
+extension StoreKit2TransactionFetcher {
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var verifiedTransactionsJWS: [String] {
+        get async {
+            return await StoreKit.Transaction.all
+                .compactMap { $0.verifiedStoreTransaction?.jwsRepresentation }
+                .extractValues()
+        }
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var subscriptionStatusBySubscriptionGroupId: [String: [Product.SubscriptionInfo.Status]] {
+        get async {
+            #if swift(>=5.9)
+            if #available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *) {
+                return await StoreKit.Product.SubscriptionInfo.Status.all
+                    .extractValues()
+                    .reduce(into: [:]) {result, value in
+                        result[value.groupID] = value.statuses
+                    }
+            } else {
+                // `StoreKit.Product.SubscriptionInfo.Status.all` is only available starting in iOS 17.0
+                // For previous versions, we retrieve all the previously purchased transactions,
+                // and fetch the subscription status only once per subscription group.
+                var transactionBySubscriptionGroup: [String: StoreKit.Transaction] = [:]
+                for await transaction in StoreKit.Transaction.all {
+                    if let verifiedTransaction = transaction.verifiedTransaction,
+                       let subscriptionGroup = verifiedTransaction.subscriptionGroupID,
+                       !transactionBySubscriptionGroup.keys.contains(subscriptionGroup) {
+                        transactionBySubscriptionGroup[subscriptionGroup] = verifiedTransaction
+                    }
+                }
+
+                let subscriptionGroups = await withTaskGroup(
+                    of: Optional<(String, Product.SubscriptionInfo.Status)>.self,
+                    returning: [String: [Product.SubscriptionInfo.Status]].self) { taskGroup in
+
+                    for (subscriptionGroup, transaction) in transactionBySubscriptionGroup {
+                        taskGroup.addTask {
+                            if let status = await transaction.subscriptionStatus {
+                                return (subscriptionGroup, status)
+                            }
+                            return nil
+                        }
+                    }
+
+                    return await taskGroup.reduce(
+                        into: [String: [Product.SubscriptionInfo.Status]]()
+                    ) { result, value  in
+                        if let value = value {
+                            result[value.0] = [value.1]
+                        }
+                    }
+                }
```

Comment:
Tried using a task group to parallelize fetching the `subscriptionStatus` for all transactions, @NachoSoto care to double check?